ursula_cli/ursula_inventory.py

```diff
@@ -0,0 +1,195 @@
+from ansible.parsing.dataloader import DataLoader
+from ansible.plugins import vars_loader
+from ansible.utils.vars import combine_vars
+from ansible.utils.vars import load_extra_vars
+from ansible.utils.vars import load_options_vars
+from ansible.vars import VariableManager
+
+LOG = logging.getLogger(__name__)
+
+
+def deep_update_dict(d, u):
+    for k, v in u.iteritems():
+        if isinstance(v, collections.Mapping):
+            r = deep_update_dict(d.get(k, {}) or {}, v)
+            d[k] = r
+        else:
+            d[k] = u[k]
+    return d
+
+
+class UrsulaInventory(Inventory):
+
+    def parse_inventory(self, host_list):
+        ungrouped = Group('ungrouped')
+        all = Group('all')
+        all.add_child_group(ungrouped)
+
+        self.groups = dict(all=all, ungrouped=ungrouped)
+
+        # Ensure basedir is inside the directory
+        host_list = os.path.join(self.host_list, "hosts")
+        self.parser = get_file_parser(host_list, self.groups, self._loader)
+        vars_loader.add_directory(self.basedir(), with_subdir=True)
+
+        self._vars_plugins = [x for x in vars_loader.all(self)]
+
+        # our default vars plugin
+        default_vars = self._get_defaults()
+
+        # set group vars from group_vars/ files and vars plugins
+        for g in self.groups:
+            group = self.groups[g]
+            group.vars = combine_vars(group.vars,
+                                      self.get_group_variables(group.name))
+            if default_vars:
+                group_vars = group.vars
+                group.vars = deep_update_dict(default_vars, group_vars)
+
+            self.get_group_vars(group)
+
+        # get host vars from host_vars/ files and vars plugins
+        for host in self.get_hosts(ignore_limits_and_restrictions=True):
+            host.vars = combine_vars(host.vars,
+                                     self.get_host_variables(host.name))
+            self.get_host_vars(host)
+
+    def _get_defaults(self):
+        p, cfg_path = load_config_file()
+        defaults_file = get_config(p, DEFAULTS, 'var_defaults_file',
+                                   'ANSIBLE_VAR_DEFAULTS_FILE', None)
+        if not defaults_file:
+            return None
+
+        ursula_env = self.host_list
+        defaults_path = os.path.join(ursula_env, defaults_file)
+        if os.path.exists(defaults_path):
+            with open(defaults_path) as fh:
+                return yaml.safe_load(fh)
+        return None
+
+
+def init_logfile():
+    config = ConfigParser()
+    config.read('ansible.cfg')
+
+    try:
+        cfg_log = config.get('defaults', 'log_path')
+    except (NoOptionError, NoSectionError):
+        cfg_log = None
+
+    if cfg_log:
+        logfile = os.path.expanduser(cfg_log)
+    else:
+        logfile = 'ursula.log'
+
+    if not os.path.exists(logfile):
+        with open(logfile, 'a'):
+            os.utime(logfile, None)
+
+
+def _initialize_logger(level=logging.DEBUG, logfile=None):
```

Comment:
That's true, but the library is still available for you to use, which you can do without error:

Setuptools still installs the entire app to `site-packages`, and the script that is created in `/usr/local/bin` or wherever is still just importing `ursula_cli.shell` and executing `main()`.  Since these are meant to go hand-in-hand, I'd get rid of the duplicated stuff, maintain one function each and merely import what's written.

```
$ pip install git+https://github.com/blueboxgroup/ursula-cli.git
... snipped ...


(tmp-98051c532d6b023f) ✔ ~/Projects/Python/virtualenvs/tmp-98051c532d6b023f
12:45 $ python -c "from ursula_cli.shell import _initialize_logger, init_logfile"
```